### ISSUE TITLE
Deprecations in github CI

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental#432]
Several deprecations are mentioned on our CI jobs. We should tackle them before they are fully deprecated and jobs start to fail.

Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: slackapi/slack-github-action

The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Comment by Itxaka** [comment]
under the elemental-operator there are actions that have no support yet for the deprecations:
 - engineerd/setup-kind
 - golangci/golangci-lint-action
 - mikefarah/yq (Fixed but waiting release)

**Comment by Itxaka** [comment]
setup-kind has no updates, sent a PR: https://github.com/engineerd/setup-kind/pull/197

**Comment by Itxaka** [comment]
elemental-operator:
 - anchore/sbom-action has a patch now, waiting for a release > 0.12.0

**Comment by Itxaka** [comment]
elemental-operator:
 - golangci-lint-action also seems to have been fixed, but no release yet.

**Comment by Itxaka** [comment]
@davidcassany IMO this should not have a milestone as its not really needed by that date. Nice to have ASAP yes, but not a such closer date. Its a pita to go over all actions ;)

**Comment by davidcassany** [comment]
Yes agree, the milestone was essentially added to avoid touching it before the 2.6.9 release.

In fact, my only concern is about running CI refactors while go/no-go tests for a release are expected to happen. However I think most, it not all, of the current deprecations are happening in non e2e tests and UI tests.

Removing the milestone.

**Comment by Itxaka** [comment]
For elemental-cli:
 - PR: https://github.com/rancher/elemental-cli/pull/345
 - golangci-lint-action also seems to have been fixed, but no release yet.

**Comment by Itxaka** [comment]
elemental-operator:
 - https://github.com/rancher/elemental-operator/pull/234

**Comment by Itxaka** [comment]
toolkit:
 - https://github.com/rancher/elemental-toolkit/pull/1639

**Comment by Itxaka** [comment]
once the toolkit PR is merged only the setup-kind action will be left from our main repos. We may need to move this to the blocked column. I pinged the main contributor for the repo in my PR to see if they are willingto merge and release a new version or we need to fork it and maintain it. I hope its done by summer 2023....

**Comment by kkaempf** [comment]
Have we looked at [renovatebot](https://github.com/renovatebot) to automate such updates ?

**Comment by Itxaka** [comment]
> Have we looked at [renovatebot](https://github.com/renovatebot) to automate such updates ?

checking....

**Comment by kkaempf** [comment]
Mostly done. Will open individual issues as needed.

**Comment by Itxaka** [comment]
@kkaempf just a heads up, patch for setup-kind action was closed as there is no intention of updating it. Thankfully kind is now available preinstalled on the ubuntu-22 nodes so there is no need for the install action anymore :)
